### PR TITLE
Spending & SpendingMonth Frontend Tests

### PR DESCRIPTION
**Pull request by msabas1**
Create frontend tests for Spending & SpendingMonth pages

### CODE REVIEW

File: cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/SpendingMonthPage.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        if(button != null) {
+            waitForElement(button, 10).click();
+        }
+        else {
+            throw new IllegalArgumentException("Button '" + name + "' does not exist.");
+        }
+
+        // give time for page to load
+        try {
+            Thread.sleep(2000);
```
Comment (achuggi1) [inline cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/SpendingMonthPage.java]: Low impact: Using a 2 second sleep could drastically increase testing times over the course of a full project.
Suggestion: Use the waitForElement method for an element that should always be on the page.

**Review by achuggi1 — CHANGES_REQUESTED** [review]
Use of long wait time Thread.sleep methods

**Review by jmensah365 — DISMISSED** [review]

File: cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/SpendingMonthPage.java (hunk 1 above)
Comment (msabas1) [reply cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/SpendingMonthPage.java]: Updated, all occurrences of try{sleep} are removed, only waitForElements being used in place

File: cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        User user = new User(UserType.PERSISTANT, "frontend.tests@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    @And("I am on the SpendingMonth page")
+    public void iAmOnTheSpendingMonthPage() {
+        this.spendingMonthPage.clickTab();
+        this.spendingPage.clickButton("See Current Month");
+        navigator.navigateTo(Navigator.PGNAME_SPENDINGMONTH);
+    }
```
Comment (Bryar-Frank) [inline cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java]: Medium Priority: I don't think it is good practice to use `this.<var>` in Step Definitions. Should just be able in this instance to use `spendingMonthPage.clickTab();`
`spendingPage.clickButton("See Current Month");`

File: cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java
Comment (Bryar-Frank) [inline cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java:1]: Make a single comment about `this`

**Review by Bryar-Frank — CHANGES_REQUESTED** [review]

File: cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java
Comment (msabas1) [reply cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java:1]: Updated code to remove 'this' from unnecessary occurrences in code

File: cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java (hunk 2 above)
Comment (msabas1) [reply cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java]: Updated code to remove 'this' from unnecessary occurrences in code

**Review by Bryar-Frank — APPROVED** [review]

**Review by jmensah365 — APPROVED** [review]

**Review by achuggi1 — APPROVED** [review]
LGTM

**Pull request merged**